src/osd/OSD.cc

```diff
@@ -7259,7 +7259,12 @@ void OSD::ms_fast_dispatch(Message *m)
     tracepoint(osd, ms_fast_dispatch, reqid.name._type,
         reqid.name._num, reqid.tid, reqid.inc);
   }
-  op->osd_parent_span = tracing::osd::tracer.start_trace("op-request-created");
+
+  if (m->otel_trace.IsValid()) {
```

Comment:
don't you need to check that `m->otel_trace` is not null?

Reply:
> don't you need to check that `m->otel_trace` is not null?

no. the `otel_trace` in the msg is not a pointer. it's the trace info which initialized to noop span context. in case we recieve an otel trace info in the msg protocol, we decode it to that span context.